src/handlers.rs

```diff
@@ -785,6 +786,27 @@ async fn try_relay_delivery(
             }
         }
         Err(e) => {
+            // Check if this is a missing public key error (offline peer scenario)
+            if let RelayError::CryptoError(CryptoError::EncryptionFailed(msg)) = &e {
+                if msg.contains("Public key not found") {
+                    // User-friendly message for offline peer without public key
+                    ui_logger.log(format!(
+                        "🔐 Cannot send secure message to offline peer '{to_name}'"
```

Comment:
Please use the Icon type from types.rs for icons so that it works on Windows too.